modules/gdscript/gdscript_parser.cpp

```diff
@@ -978,8 +978,11 @@ GDScriptParser::Node *GDScriptParser::_parse_expression(Node *p_parent, bool p_s
 						return NULL;
 					}
 					Node *n = _parse_expression(arr, p_static, p_allow_assign, p_parsing_constant);
-					if (!n)
+					if (!n) {
 						return NULL;
+					} else if (n->type == Node::TYPE_IDENTIFIER) {
```

Comment:
Is the `if` really necessary? Wouldn't things work just fine if every node gets its type reduced?

Also, shouldn't similar code be added to dictionary parsing below?